docs/contributing/managing-change.md

```diff
@@ -171,3 +171,11 @@ Keep the old name in the selector list, and mark it as deprecated.
   foo: bar;
 }
 ```
+
+### The `_warning` mixin and allowing users to suppress warnings
+
+In the above examples we've used `@include _warning(...)` instead of the native sass `@warn` at-rule. We do this instead of using `@warn` because it gives users the option to suppress deprecation warnings by interacting with the `$govuk-suppressed-warnings` map.
+
+The `_warning` mixin takes 2 parameters: a warning key (`$key`) and a warning message (`$message`), checks if `$key` isn't already included in `$govuk-suppressed-warnings` and if it isn't, outputs a warning via `@warn` using `$message` plus a notice to add the `$key` to `$govuk-suppressed-warnings` if they want to silence the warning.
```

Comment:
```suggestion
If users want to silence the warning, the `_warning` mixin looks at 2 parameters, a warning key (`$key`) and a warning message (`$message`), and checks whether `$key` is included in `$govuk-suppressed-warnings`. If `$key` is not included, the `_warning` mixin outputs a warning through `@warn` using `$message` and adds a notice to add the `$key` to `$govuk-suppressed-warnings`.```

Reply:
Please check whether I've changed the meaning of this.

Reply:
I don't think this changes the meaning, however looking at it now that you've tidied it up I think the last sentence needs a bit more detail to make it clear that the `@warn` output is the passed `$message` with some instructions on how to silence it:

```
If `$key` is not included, the `_warning` mixin outputs a warning through `@warn` using `$message` with the addition of notice to add the `$key` to `$govuk-suppressed-warnings` as the warning text.
```


Let me know what you think.

Reply:
yup, that is better.

Reply:
I think although it's mostly correct this has got a little muddled, as the mixin internally works the same whether the user wants to silence the warnings or not.

Try as I might, I also can't get my head around what this sentence means:

> If `$key` is not included, the `_warning` mixin outputs a warning through `@warn` using `$message` with the addition of notice to add the `$key` to `$govuk-suppressed-warnings` as the warning text.

What about something like:

> The `_warning` mixin accepts 2 parameters, a warning key (`$key`) and a warning message (`$message`). The warning message is only outputted if the `$key` is not already in the `$govuk-suppressed-warnings` list. Once the message has been outputted, `$key` is then added to the `$govuk-suppressed-warnings` list.
> 
> This means that only one warning be outputted for a given `$key`, and that users can suppress warnings by adding them to the `$govuk-suppressed-warnings` list before they include GOV.UK Frontend.

Reply:
That's true actually I missed that. The `_warning` mixin has nothing to do with the user wanting to suppress warnings as its a private mixin 🙃  Users interact with this functionality via `$govuk-suppressed-warnings`.

This makes sense to me. @claireashworth Do you have an opinion?

Reply:
As for that line you didn't understand, this is trying to explain that the `@warn` within `_warning` is going to output `$message` plus a little addendum telling users they can suppress warnings. It's also contextual to the previous sentence which explains how `$key` and `$govuk-suppressed-warnings` interact.

I'm starting to wonder if we include this paragraph at all since I feel like it's trying to do the job that the sassdoc for `$govuk-suppressed-warnings` and `_warning` are already doing. Do either of you object to me removing this paragraph altogether, potentially replacing it with something like:

>You can read more about how `$govuk-suppressed-warnings` and `_warning` work by reading their respective sassdocs.

Ideally I'd link out to the docs page but it's not live yet. Maybe something to do post release?

Reply:
The docs for `_warning` won't end up in Frontend Docs because we don't include private things in our API reference, but otherwise that sounds good to me.

Reply:
I've scooped it out. Let me know what you reckon now.